src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java

```diff
@@ -21,19 +21,27 @@
 @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 @Slf4j
 public class VMWorkloadInventoryCalculator extends AbstractVMWorkloadInventoryCalculator implements Calculator<Collection<VMWorkloadInventoryModel>> {
-
     @Override
     public Collection<VMWorkloadInventoryModel> calculate(String cloudFormsJson, Map<String, Object> headers) {
+
         manifestVersion = getManifestVersion(cloudFormsJson);
         jsonParsed = JsonPath.parse(cloudFormsJson);
         scanRunDate = getScanRunDate();
 
         List<Map> vmList = readListValuesFromExpandedEnvVarPath(VMPATH, null);
 
-        return vmList.stream().map(e -> {
-            e.put("_analysisId", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString());
-            return e;
-        }).map(this::createVMWorkloadInventoryModel).collect(Collectors.toList());
+        List<VMWorkloadInventoryModel> vmWorkloadInventoryModels = vmList.stream()
+                .peek(e -> {
+                    e.put("_analysisId", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString());
+                    e.put("vmEmsCluster", readValueFromExpandedEnvVarPath(VMEMSCLUSTERPATH, e));
+                    e.put("ems_cluster_id", readValueFromExpandedEnvVarPath(EMSCLUSTERIDPATH, e));
+                })
+                .peek(e -> log.debug("------- Treating Analysis {} VM :{} from {} : ", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString(), vmList.indexOf(e), vmList.size()))
```

Comment:
`if (log.isDebugEnabled())` will avoid the `vmList.indexOf(e)` evaluation

Reply:
I dont understand

Reply:
`vmList.indexOf(e)` is executed every time, no matter the log level.

Reply:
+1